contracts/src/Surge.sol

```diff
@@ -101,87 +70,50 @@ contract Surge is ERC721, ReentrancyGuard, Ownable, ERC721Enumerable {
+    function presaleMint(uint256 _amountOfTokens, bytes32[] calldata _merkleProof)
         external
         payable
         nonReentrant
-        isPresaleActive
-        maxMint(_amountOfTokens)
+        verifyMint(_amountOfTokens)
         isEnoughEth(_amountOfTokens)
-        isInPresale
-        hasMintedMaxPresale
     {
-        for (uint256 i = 0; i < _amountOfTokens; i++) {
-            uint256 newTokenId = _tokenIds.current() + 1;
-            require(newTokenId <= MAX_TOKENS, "No available tokens to mint");
-            _safeMint(msg.sender, newTokenId);
-            _tokenIds.increment();
-        }
-        if (balanceOf(msg.sender) == MAX_PER_USER) {
-            _presaleMinted[msg.sender] = true;
-        }
-    }
-
-    //gift minting
-    function giftMint(address[] calldata _receivers) external nonReentrant onlyOwner {
-        uint256 totalReceivers = _receivers.length;
-        for (uint256 i = 0; i < totalReceivers; i++) {
-            //checks if there is enough reserved token for gifting left
-            require(totalGiftMints <= MAX_RESERVED_TOKENS, "No available tokens for gifting");
-            require(balanceOf(_receivers[i]) + 1 <= MAX_PER_USER, "Wallet has max number of tokens allowed");
-            totalGiftMints++;
-            uint256 newTokenId = _tokenIds.current() + 1;
-            _safeMint(_receivers[i], newTokenId);
-            _tokenIds.increment();
-        }
-    }
-
-    //getter for tokens owned by a user
-    function getTokens(address _owner) external view returns (uint256[] memory) {
-        uint256 totalCount = balanceOf(_owner);
-        uint256[] memory tokenIds = new uint256[](totalCount);
+        require(presaleIsActive, "Presale is currently not active");
+        require(_presaleApproved[msg.sender], "You are not in the pre-sale");
+        require(!_presaleMinted[msg.sender], "You have already minted your tokens for the presale");
+        bytes32 leaf = keccak256(abi.encodePacked(_msgSender()));
+        require(MerkleProof.verify(_merkleProof, merkleRoot, leaf), 'Invalid proof!');
 
-        for (uint256 i = 0; i < totalCount; i++) {
-            tokenIds[i] = tokenOfOwnerByIndex(_owner, i);
+        _safeMint(msg.sender, _amountOfTokens);
+        if (balanceOf(msg.sender) == maxPerUser) {
+            _presaleMinted[msg.sender] = true;
         }
-
-        return tokenIds;
-    }
-
-    //ERC721 Enumerable
-    function _beforeTokenTransfer(
-        address _from,
-        address _to,
-        uint256 _tokenId
-    ) internal virtual override(ERC721, ERC721Enumerable) {
-        super._beforeTokenTransfer(_from, _to, _tokenId);
     }
 
-    function supportsInterface(bytes4 _interfaceId) public view override(ERC721, ERC721Enumerable) returns (bool) {
-        return super.supportsInterface(_interfaceId);
+    // premint function allows the owner to mint maxReserved amount
+    function preMint(uint256 _amountOfTokens) external nonReentrant onlyOwner {
```

Comment:
Why is this called `premint` rather than `mint`?

Reply:
This is a custom method I created for the owner to be able to mint 200. It is the `giftMint` function. Since the verbiage is confusing I will rename to `giftMint`.

Reply:
And let's keep the amount of tokens `owner` can mint unlimited in case we find ourselves in the rocky situation of needing to "burn" these into our wallet later on if we don't sell out. WHICH WON'T HAPPEN but better prepare... haha